src/llmq/chainlocks.cpp

```diff
@@ -25,17 +25,17 @@ namespace llmq
 {
 std::unique_ptr<CChainLocksHandler> chainLocksHandler;
 
-CChainLocksHandler::CChainLocksHandler(CChainState& chainstate, CConnman& _connman, CMasternodeSync& mn_sync, CQuorumManager& _qman,
+CChainLocksHandler::CChainLocksHandler(CChainState& chainstate, CConnman& _connman, CQuorumManager& _qman,
                                        CSigningManager& _sigman, CSigSharesManager& _shareman, CSporkManager& sporkman,
-                                       CTxMemPool& _mempool) :
+                                       CTxMemPool& _mempool, const CMasternodeSync& mn_sync) :
```

Comment:
why that? add `const` - ok, why changing order here?

Reply:
Because we place `const` args _after_ non-`const` args, see [`LLMQContext`](https://github.com/dashpay/dash/blob/27c0813c08ac7e72d564120f966b3ba1e3804ae1/src/llmq/context.h#L37-L39), [`CJContext`](https://github.com/dashpay/dash/blob/27c0813c08ac7e72d564120f966b3ba1e3804ae1/src/coinjoin/context.h#L32-L33) and [`CDSNotificationInterface`](https://github.com/dashpay/dash/blob/27c0813c08ac7e72d564120f966b3ba1e3804ae1/src/dsnotificationinterface.h#L20-L25) for examples.

Reply:
there's no clear agreement about order of const/non-const argument in methods, but this one is the closest:

> For consistency and friendliness to code generation tools, interface method input and inout parameters should be ordered first and output parameters should come last.
https://github.com/dashpay/dash/blob/e210cb734e16e0ecef9601251fe0e074ed86827e/doc/developer-notes.md

I would say that `const` argument are better to go first rather than being after non-const, thoughts?

Reply:
If we're talking about arguments that can be clearly understood as inputs/outputs, then I agree.

Though we have this middle ground with references to managers that are read-write (non-`const`) or read-only (`const`), some references are used to access functions that seem to do only read-only things even when behind the scenes they are read-write (like `CDeterministicMNManager::GetListAtTip()`, which is non-`const`).

Plus, sometimes we use `const` for `unique_ptr` refs, where we are only `const`ing the container ref but intend to do read-write activities with the underlying type. It starts becoming less clear.

Maybe it can examined at some point but I think current convention is inoffensive, not particularly misleading.

Reply:
> we use const for unique_ptr refs
there's a difference between `const unique_ptr<T>` and `const unique_ptr<const T>`

Reply:
> sometimes we use const for unique_ptr refs

indeed, but all `const unique_ptr` are just temporary workarounds until circular dependencies are broken

Reply:
IMO as long as they're there in the reasonable future, they need to be considered.

Reply:
so, still, is there any reason why you moved `const` argument from the beginning to the end? If there's anything uses _not the best code style_ that's not a reason to spread it over codebase.

Reply:
I think the placement here is fine. I think it only needs to be enforced that "out params" are at the end. Ordering of const and non-const in or inout params, is not important.